bin/common.c

```diff
@@ -57,6 +57,74 @@ static uint8_t unsafe_verify_host_fn(const char *host_name, size_t host_name_len
     return 1;
 }
 
+int write_array_to_file(const char *path, uint8_t *data, size_t length) {
```

Comment:
Doesn't seem like there are any comments on the functions in this file, but maybe this is a good time to start adding some just describing what these do, and what the inputs/outputs are?

Reply:
Hmmm maybe I should start the trend of adding doxygen comments to our common.h file 🤔.